hell/adventcalendar/2025/20/index.md

```diff
@@ -1,35 +1,111 @@
 date: 2025-12-20
 author_links:
-  - label: "Site"
-    url: "https://linktoyourblog123.com"
-    link_label: "linktoyourblog123.com"
-intro: "<p>Short introductory text</p>"
+  - label: "blog"
+    url: "https://css-weblog.de"
+    link_label: "My German weblog, mostly about CSS."
+  - label: "Bluesky"
+    url: "https://bsky.app/profile/jensgro.bsky.social"
+    link_label: "jensgro.bsky.social"    
+  - label: "Mastodon"
+    url: "https://mastodon.social/@jensgro"
+    link_label: "@jensgro@mastodon.social"
+  - label: "Codepen"
+    url: "https://codepen.io/jensgro"
+    link_label: "jensgro"
+intro: "<p>Forms, huge or compact, are one of the important building blocks of the web. Without forms the web would be a one-way street. With HTML5 we received an optimized set of form elements. HTML5 turned into a Recommendation in 2014. But until now web developers have to struggle with forms.</p>"
 image: "advent25_20"
 ---
-Some text.
-Some text.
 
-Some text. Some text.
+Forms were likely one of the reasons why browser vendors joined forces in the WHATWG in 2004. They felt that HTML standardization was heading into the wrong direction and wanted more practical relevance. While this may be an oversimplification, if true, it highlights the failure of the WHATWG (i.e., the browser vendors) and, subsequently, the W3C. Although the newly standardized form elements and features all point into the right direction, they are incomplete and unfinished. The fact that this is still the case, even more than ten years after HTML5 became a recommendation, is alarming. I will cencentrate in this article on missing elements, inconsistent behaviour and the problems with styling.
 
-## Heading
+## New Form Elements
 
-Some text.
+The WHATWG introduced several new form elements and attributes into the HTML specification, e.g. the email input, telephone input, date field, and range slider. Most of these are simply new types of the input element. This approach was clever — a great example of progressive enhancement. If a browser hasn’t implemented a particular input type yet, a plain text field serves as a fallback. Developers can then use JavaScript if needed. However, when the new elements are supported, JavaScript becomes unnecessary.
 
-## Heading
+This progressive-enhancement model works well for simple inputs. So these new elements are intentionally very simple – essentially shortcuts for single-value fields. They combine regular expressions for validation, an ARIA role, and sometimes trigger a specialized virtual keyboard on mobile phones and tablets. They are similar to web components based on the input element, but are integrated directly into the browser, so no JavaScript is needed for the element to appear. And because they are standardized, the code is consistent across implementations. These are very simple elements.
```

Comment:
> These are very simple elements.

You already said they're simple at the beginning of this paragraph. Is there a reason you're repeating it? If yes, the reason of your emphasis on the simplicity isn't clear.